### DISCUSSION TITLE
how to config  in mbed ?

### DISCUSSION THREAD

**Discussion by Layty** [cesanta/mongoose#2578] [Ask questions, get answers] [answered]
how to config  in mbed ?  

# Environment

- mongoose version:  master 
- Compiler/IDE and SDK:  mbed-os-6 master
- Target hardware/board:  nxp RT1051
- Target RTOS/OS (if applicable): _?_ mbed

**Comment by scaprile** [comment, marked as answer]
We don't have examples for mbed-os nor for RT1051
Mongoose runs on socket-based stacks
Have you tried ? Did you find any issues ? Can you post your errors/warnings ? What is your compiler ? What is your runtime library ? Does it use newlib ? Does it use armgcc ?

**Reply by Layty** [reply]
I have use lwip and not use mbed-os cpp api,  then the demo can accept just once, and next client can not connect to the server on the board.  I think  there is some config is not right.  I use arm-gcc . 

Now  I have to pause this work.  thank you. 

is there any to help me to guide how to use the lib with out raw socket api, just use overide send and recv api  ? I want use the custom socket api

**Reply by scaprile** [reply]
If there is an actual issue, please open an issue, _honor the issue template_ so we can know your scenario and reproduce the issue if necessary, and let us know _exactly_ what you are doing,
Mongoose does NOT work over lwip RAW API, it does require a known socket API or a supported driver for its built-in TCP/IP stack. That includes CMSIS-driver, you might try that.
